frontend/src/metabase/dashboard/reducers.js

```diff
@@ -142,12 +143,13 @@ const dashboards = handleActions(
         ),
     },
     [Dashboards.actionTypes.UPDATE]: {
-      next: (state, { payload }) =>
-        assocIn(
-          state,
-          [payload.dashboard.id, "collection_id"],
-          payload.dashboard.collection_id,
-        ),
+      next: (state, { payload }) => {
+        return produce(state, draftState => {
+          const draftDashboard = draftState[payload.dashboard.id];
+          draftDashboard.collection_id = payload.dashboard.collection_id;
+          draftDashboard.collection = payload.dashboard.collection;
```

Comment:
handle moving dashboard to another collection type (public/personal) as our components rely on `dashboard.collection` property.